vm/src/builtins/type.rs

```diff
@@ -426,11 +428,37 @@ impl Py<PyType> {
     /// so only use this if `cls` is known to have not overridden the base __subclasscheck__ magic
     /// method.
     pub fn fast_issubclass(&self, cls: &impl Borrow<crate::PyObject>) -> bool {
-        self.as_object().is(cls.borrow()) || self.mro.iter().any(|c| c.is(cls.borrow()))
+        self.as_object().is(cls.borrow()) || self.mro.read().iter().any(|c| c.is(cls.borrow()))
+    }
+
+    pub fn mro_map_collect<F, R>(&self, f: F) -> Vec<R>
```

Comment:
and defined here again.
`&Py<PyType>` derefs to `PyType`. So this one probably is redundant.

Does this method can be `mro_map`(or map_mro) without collect?
Or is it not possible anymore due to RwLock?

Reply:
Yeah, the iterator needs to hold the lock on `mro` so the only way to have a helper method is to collect the result. I think it's clumsy but it's the best way I could find :/

Reply:
I think it's not redundant, it is used here:
```rust
#[pyclass]
impl Py<PyType> {
    ...
    #[pymethod]
    fn mro(&self) -> Vec<PyObjectRef> {
        self.mro_map_collect(|cls| cls.to_owned().into())
    }
}
```

The `PyType` implementation discards the `Py` wrapper, which is still needed in this case. Is there a better way to write this so that the `PyType::mro_map_collect` can be used?

Reply:
Woops, I missed https://github.com/RustPython/RustPython/pull/5325#discussion_r1605958799
^ Ignore that